### PR TITLE
nla_interval initial review

### PR DESCRIPTION
**Pull request by NikolajBjorner**

### CODE REVIEW

File: src/math/lp/nla_intervals.h [hunk 1]
```diff
@@ -229,221 +228,310 @@ class intervals {
...
+        }
+
+        interv a;
+        set_interval_for_scalar(a, e.coeff());
+        TRACE("nla_intervals_details", tout << "a = "; display(tout, a); );
+        for (const auto& ep : e) {
+            interv b = interval_of_expr<wd>(ep.e(), ep.pow());
+            TRACE("nla_intervals_details", tout << "ep = " << ep << ", "; display(tout, b); );
+            interv c;
+            interval_deps_combine_rule comb_rule;
```
Comment (NikolajBjorner) [inline src/math/lp/nla_intervals.h]: There was not a version of this function that ignored dependencies.
Shouldn't this code be specialized to the value of wd?

File: src/math/lp/nla_intervals.h [hunk 2]
```diff
@@ -229,221 +228,310 @@ class intervals {
...
+    interval interval_of_sum(const nex_sum& e) {
+        TRACE("nla_intervals_details", tout << "e=" << e << "\n";);
+        interv i_e = interval_of_sum_no_term<wd>(e);
+        if (e.is_a_linear_term()) {
+            SASSERT(e.is_sum() && e.size() > 1);
+            interv i_from_term;
+            if (interval_from_term<wd>(e, i_from_term)) {
+                interv r = intersect<wd>(i_e, i_from_term);
+                TRACE("nla_intervals_details", tout << "intersection="; display(tout, r) << "\n";);
+                if (is_empty(r)) {
```
Comment (NikolajBjorner) [inline src/math/lp/nla_intervals.h]: Not sure why to have an assert here. Surely odd, but what if if we have a sum where variables have empty intervals?

File: src/math/lp/nla_intervals.h [hunk 3]
```diff
@@ -229,221 +228,310 @@ class intervals {
...
+        rational a, b;
+        lp::lar_term norm_t = expression_to_normalized_term(&e.to_sum(), a, b);
+        lp::explanation exp;
+        if (m_core->explain_by_equiv(norm_t, exp)) {
+            set_zero_interval(i);
+            TRACE("nla_intervals", tout << "explain_by_equiv\n";);
+            return true;
+        }
+        lpvar j = find_term_column(norm_t, a);
+        if (j + 1 == 0)
```
Comment (NikolajBjorner) [inline src/math/lp/nla_intervals.h]: elsewhere in Z3 we use a "null_theory_var" or "null_bool_var". Why not have a "null_dlvar"? Then don't use these checks for -1, but compare to "null_dlvar" instead.

Comment (levnach) [reply src/math/lp/nla_intervals.h]: Do you meas null_plvar?

File: src/math/lp/nla_intervals.h (hunk 1 above)
Comment (levnach) [reply src/math/lp/nla_intervals.h]: Yes, it should. I was forgetting to add it.

Comment (levnach) [reply src/math/lp/nla_intervals.h]: Fixed now.

**Pull request merged**